lib/pbench/agent/utils.py

```diff
@@ -188,3 +190,40 @@ def collect_local_info(pbench_bin):
         hostdata[arg] = cp.stdout.strip() if cp.stdout is not None else ""
 
     return (version, seqno, sha1, hostdata)
+
+
+# From https://stackoverflow.com/questions/106179/regular-expression-to-match-dns-hostname-or-ip-address
+_allowed = re.compile(
+    r"^(([a-zA-Z]|[a-zA-Z][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z]|[A-Za-z][A-Za-z0-9\-]*[A-Za-z0-9])$"
+)
```

Comment:
I have several suggestions, here:

- Add `flags=re.IGNORECASE` to the `re.compile()` call so that you don't need to include both letter ranges.
- Use `re.fullmatch()` and omit the anchors.
- Use non-capturing groups.
- Based on the way that you are actually using this RE, you don't need to look for dots in it.
- RFC 1123 states:  "the restriction on the first character is relaxed to allow either a letter or a digit" which has the side effect of allowing your algorithm to match either dotted-decimal quartets or FQDNs.

So, I think you can simplify this to just:
```
_allowed = re.compile(r"[A-Z0-9](?:[A-Z0-9\-]{0,61}[A-Z0-9])?", flags=re.IGNORECASE)
```
That is, a case-blind comparison seeking a letter-or-digit, optionally followed by a sequence of zero to 61 letter-or-digit-or-hyphens followed by a letter-or-digit.

Reply:
I'll address in PR #2207.